src/bn_txns.erl

```diff
@@ -206,6 +254,32 @@ save_transactions(Height, Transactions, Ledger, Chain, #state{
     bn_db:batch_put_follower_height(Batch, DefaultCF, Height),
     rocksdb:write_batch(DB, Batch, [{sync, true}]).
 
+-spec get_historic_balance(Height :: pos_integer(), Address :: binary()) ->
+    {ok, Entry :: blockchain_ledger_entry_v1:entry()} | {error, term()}.
+get_historic_balance(Height, Address) ->
+    #state{balances=BalancesDB} = get_state(),
+    {ok, EntryBin} = rocksdb:get(BalancesDB, erlang:term_to_binary({Address, Height}), []),
+    Entry = blockchain_ledger_entry_v1:deserialize(EntryBin),
+    {ok, Entry}.
+
+save_balances(Ledger, Height, #state{balances=BalancesDB}) ->
+    {ok, Batch} = rocksdb:batch(),
+    Accounts = blockchain_ledger_v1:snapshot_raw_accounts(Ledger),
```

Comment:
This is massively expensive given that you're iterating over large numbers of accounts to store the potentially same balance again. I would use ledger commit hooks to notice when accounts have changed and storing just the changed accounts then? 

It would make looking up the balance of an account at any given height a bit harder since you're have to iterate the balances table backwards to find the last entry for a given account.

Reply:
will look into commit hooks, looks like etl has some examples of it?

Reply:
Yes.